### PR TITLE
Deve: Budget App Capstone Project

### PR DESCRIPTION
**Pull request by Cosmosha**
**The following requirements were implemented;**

**Splash screen**

- A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.
Sign up and log in pages

- The user should be able to register in the app with full name, email and password (all mandatory).
- The user can log into the app using email and password.
- If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

**Home page (categories page)**

- When the user logs in, they are presented with the categories page.
- For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.

**Transactions page**

- For a given category, the list of transactions is presented, ordered by the most recent.

**"Add a new category" page**

**"Add a new transaction" page**

**The user fills out a form to create a new transaction with:**
- name (mandatory)
- amount (mandatory)
- categories (mandatory at least one)
- The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
- When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.
- 
**Testing requirements**
Create unit and integration tests for all the most important components of your RoR application.

**Technical requirements**
- You should use Postgres as your database.
- You should use devise for authentication.
- You should validate all user input to make sure that anyone with bad intentions cannot compromise your app.
- You can use a view template engine of your choice (.erb, .slim, .haml).
- The project should be deployed and accessible online.
- Your database schema should reflect the following structure:

### CODE REVIEW

File: app/views/entities/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+                <h4 class="card-title">Payment</h4>
+                <h4 class="card-subtitle mb-2">Total: $<%= @group.entities.sum(&:amount) %></h4>
+            </div>
+        </div>
+    </div>
+</div>
+
+<footer>
+    <%= link_to "add new transaction", new_group_entity_path(group_id: @group.id) %>
+</footer>
```
Comment (codepantha) [inline app/views/entities/index.html.erb:48]: - [ ] Please follow the design guideline for the transactions page on this project. Kindly take note of itemized areas where you need to make changes below 👇 
1. For the transaction cards, please implement the exact layout (composition and space between items) in the design guideline. Also ensure to follow the font-weight and text colors 👍 
2. Please ensure that the total payment of the transactions is at the top of the page and not at the bottom just like in the design guideline 👍 
3. Please style the button according to the design guideline:
    - Please note that the background color used should be `#5fb523`
### Design Guideline
image

### Your Design
image

File: app/views/groups/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            <h4><%= group.created_at.strftime('%B %d, %Y %I:%M %p') %></h4>
+          </div>
+        </div>
+      <% end %>
+    <% end %>
+  </div>
+
+  <footer>
+    <%= link_to "Add a new category", new_group_path %>
+  </footer>
```
Comment (codepantha) [inline app/views/groups/index.html.erb:37]: - [ ] Please follow the design guideline for the groups or categories page on this project. Kindly take note of itemized areas where you need to make changes below 👇 
1. For the group cards, please implement the exact layout (composition and space between items) in the design guideline. 
   - Please ensure that the background color for the group cards is same as in the design guideline 🙏 
   - Also ensure to follow the font-weight and text colors especially for the pricing 👍 
3. Please style the button according to the design guideline:
    - Please note that the background color used should be `#5fb523`
### Design Guideline
image

### Your Design
image

File: app/views/entities/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+    <div class="field">
+        <%= f.label :group %><br />
+        <%= f.collection_select :group_id, current_user.groups, :id, :name, { class: "options form-control form-control-lg", selected: @group.id } %>
+    </div>
+
+    <div class="actions">
+        <%= f.submit "Add Entity", class: "btn btn-primary btn-lg btn-block splashBtn" %>
+    </div>
+<% end %>
```
Comment (codepantha) [inline app/views/entities/new.html.erb:30]: - [ ] Please style the form input elements and the submit button according to the design guidelines. Please note that the input elements should be wider and take up almost all the available width 👍 
- [ ] Please apply same changes to the form to create a new group on the `groups/new.html.erb` page🙏 
### Design Guideline
image

### Your Design
image

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @Cosmosha 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Project deployed and accessible online 📌 
✔️ Good commit history
✔️ No linter errors 🟢 

## Required Changes ♻️
_Check the comments under the review._
- [ ] Well done for writing unit tests for your application. However, you also need to write integration tests for the most important components of your application as stated in the project requirements 👇 
image
- [ ] Please ensure that your demo video contains a highlight about a piece of the code that you find interesting or are proud of making 🙏 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/entities/index.html.erb (hunk 1 above)
Comment (Cosmosha) [reply app/views/entities/index.html.erb:48]: HI @codepantha ,
thank you for taking the time to review my project, however, I believe the design had a Gift Credit that was displayed at the top. however, my design displays the total amount of item price entered, that's why I decided to give it the design to give the user a brief understanding of the expenses. However, I will do well to implement the needed changes you mentioned. Thanks once again.

Comment (codepantha) [reply app/views/entities/index.html.erb:48]: @Cosmosha I like how you tried to make your design unique 👏 That is commendable, however, it is better to stick to the design while treating it as a user requirement.

**Pull request merged**